### PR TITLE
MCOL-641 1. Add support for int128_t in ParsedColumnFilter.

### PR DESCRIPTION
**Pull request by tntnatbry**
2. Set Decimal precision in SimpleColumn::evaluate().
3. Add support for int128_t in ConstantColumn.
4. Set IDB_Decimal::s128Value in buildDecimalColumn().
5. Use width 16 as first if predicate for branching based on decimal width.

### CODE REVIEW

**Comment by tntnatbry** [comment]
Functional tests for DELETE are passing after this commit. This also addresses most of the comments in https://github.com/drrtuy/mariadb-columnstore-engine/pull/10 and https://github.com/drrtuy/mariadb-columnstore-engine/pull/11.

File: dbcon/joblist/jlf_common.cpp [hunk 1]
```diff
@@ -333,7 +333,9 @@ string extractTableAlias(const SSC& sc)
 //------------------------------------------------------------------------------
 CalpontSystemCatalog::OID isDictCol(const CalpontSystemCatalog::ColType& colType)
 {
-    if (datatypes::Decimal::isWideDecimalType(colType)) return 0;
+    if (datatypes::Decimal::isWideDecimalType(colType) ||
+        (colType.colDataType == CalpontSystemCatalog::BINARY &&
+         utils::isNarrow(colType.colWidth))) return 0;
```
Comment (drrtuy) [inline dbcon/joblist/jlf_common.cpp]: Binary 16 is not a dict also

Comment (tntnatbry) [reply dbcon/joblist/jlf_common.cpp]: Ok. I have addressed this in the new PR: https://github.com/drrtuy/mariadb-columnstore-engine/pull/13

**Comment by tntnatbry** [comment]
PR https://github.com/drrtuy/mariadb-columnstore-engine/pull/13 has been opened that rebases changes in this PR on top of MCOL-641_1_5. Closing this.

**Pull request closed without merging**